### DISCUSSION TITLE
collapse other directories,  but only open the directory of current buffer.

### DISCUSSION THREAD

**Discussion by zztczcx** [nvim-tree/nvim-tree.lua#1929] [Q&A] [answered]
so when I jump between buffers, I only want the current buffer's nested directory to open and display in sidebar. 
I don't want to have other directories open that is not belong to my current buffer. 

the only solution I can think of is 

run :NvimTreeCollapse first, then open NvimTree.

is there other better solution?  without typing command.

**Comment by alex-courtis** [comment, marked as answer]
You could use a little bit of automation like this:
```lua
local nt_api = require("nvim-tree.api")

local function collapse_and_find(data)
  nt_api.tree.collapse_all()
  nt_api.tree.find_file(data.file)
end

vim.api.nvim_create_autocmd({ "BufEnter" }, { callback = collapse_and_find })
```

You could add some filtering of filetypes / buffer types etc. however I'll leave that as an exercise for the reader.

**Reply by zztczcx** [reply]
@alex-courtis  , Thanks for your reply.
it is almost what I want. except it needs one small improvement. 
For example, when I have two buffers. 
when I jump between them, the nvim-tree only open its corresponding directories, which is very good. 
however, if I [C-w H]  jump to the nvim-tree.  all directories collapse. 
can I keep nvim-tree open directory of the current-butter ?  (the one that displayed, I know on bufferline, there is no buffer focused). 
should I bind the function to another autocmd?

**Reply by alex-courtis** [reply]
>when I have two buffers. when I jump between them, the nvim-tree only open its corresponding directories, which is very good. however, if I [C-w H] jump to the nvim-tree. all directories collapse. 

Ideas:

```lua
  -- only collapse when we have changed to a buffer with a real file on disk
  if vim.fn.filereadable(data.file) ~= 1 then
    return
  end
```

```lua
  -- don't collapse when entering nvim-tree
  if vim.bo[data.buf].buftype == "NvimTree" then
    return
  end
```

See https://github.com/nvim-tree/nvim-tree.lua/issues/1669#issuecomment-1374708397 for more ideas.

> should I bind the function to another autocmd?

You could. There are probably better events to use than BufEnter.
